src/Java.Interop/Java.Interop/JniRuntime.JniValueManager.cs

```diff
@@ -276,16 +276,45 @@ static Type GetPeerType ([DynamicallyAccessedMembers (Constructors)] Type type)
 				if (disposed)
 					throw new ObjectDisposedException (GetType ().Name);
 
+				if (!reference.IsValid) {
```

Comment:
Though this check being first means I'm fibbing about "always checking for fundamental errors"; if the value were always null, for whatever reason, we'd never get around to "sanity checking" whether it's even *possible*. 🤔